optas/builder.py

```diff
@@ -75,20 +75,22 @@ def __init__(self, T, robots=[], tasks=[], optimize_time=False, derivs_align=Fal
 
         # Setup decision variables
         self._decision_variables = SXContainer()
+        self._parameters = SXContainer()
         for model in self._models:
             for d in model.time_derivs:
-                n = model.state_name(d)
+                n_s = model.state_name(d)
+                n_p = model.parameter_name(d)
                 if model.T is None:
                     t = T-d if not derivs_align else T
                 else:
                     t = model.T
-                self.add_decision_variables(n, model.dim, t)
+                self.add_decision_variables(n_s, len(model.opt_joint_names), t)
+                self.add_parameter(n_p, len(model.param_joint_names), t)
 
         if optimize_time:
             self.add_decision_variables('dt', T-1)
 
         # Setup containers for parameters, cost terms, ineq/eq constraints
-        self._parameters = SXContainer()
         self._cost_terms = SXContainer()
         self._lin_eq_constraints = SXContainer()
         self._lin_ineq_constraints = SXContainer()
```

Comment:
The comments potentially need updating. However, I am thinking maybe move this code block to above the `# Setup decision variables` block. Then the `SXContainer`'s are already initialized. Then you can setup the decision variables and parameters as you have it without splitting up where the `SXContainer`'s are initialized.

Reply:
Done.